pilot/pkg/networking/plugin/mixer/mixer.go

```diff
@@ -80,21 +84,57 @@ func (Plugin) OnInboundListener(in *plugin.CallbackListenerInputParams, mutable
 	return fmt.Errorf("unknown listener type %v in mixer.OnOutboundListener", in.ListenerType)
 }
 
-// OnOutboundCluster implements the Callbacks interface method.
+// OnOutboundCluster implements the Plugin interface method.
 func (Plugin) OnOutboundCluster(env model.Environment, node model.Proxy, service *model.Service, servicePort *model.Port, cluster *xdsapi.Cluster) {
 }
 
-// OnInboundCluster implements the Callbacks interface method.
+// OnInboundCluster implements the Plugin interface method.
 func (Plugin) OnInboundCluster(env model.Environment, node model.Proxy, service *model.Service, servicePort *model.Port, cluster *xdsapi.Cluster) {
 }
 
-// OnOutboundRoute implements the Callbacks interface method.
-func (Plugin) OnOutboundRoute(env model.Environment, node model.Proxy, route *xdsapi.RouteConfiguration) {
+// OnOutboundRouteConfiguration implements the Plugin interface method.
+func (Plugin) OnOutboundRouteConfiguration(in *plugin.InputParams, routeConfiguration *xdsapi.RouteConfiguration) {
 }
 
-// OnInboundRoute implements the Callbacks interface method.
-func (Plugin) OnInboundRoute(env model.Environment, node model.Proxy, service *model.Service, servicePort *model.Port,
-	route *xdsapi.RouteConfiguration) {
+// oc := BuildMixerConfig(node, serviceName, dest, proxyInstances, config, mesh.DisablePolicyChecks, false)
+// func BuildMixerConfig(source model.Proxy, destName string, dest *model.Service, instances []*model.ServiceInstance, config model.IstioConfigStore,
+
+// OnInboundRouteConfiguration implements the Plugin interface method.
+func (Plugin) OnInboundRouteConfiguration(in *plugin.InputParams, routeConfiguration *xdsapi.RouteConfiguration) {
+	forward := false
+	if in.Node.Type == model.Ingress {
+		forward = true
+	}
+
+	switch in.ListenerType {
+	case plugin.ListenerTypeHTTP:
+		var nvhs []route.VirtualHost
+		for _, vh := range routeConfiguration.VirtualHosts {
+			nvh := vh
+			var nrs []route.Route
+			for _, r := range vh.Routes {
+				nr := r
+				if nr.Metadata == nil {
+					nr.Metadata = &core.Metadata{}
+				}
+				if nr.Metadata.FilterMetadata == nil {
+					nr.Metadata.FilterMetadata = make(map[string]*types.Struct)
+				}
+				nr.Metadata.FilterMetadata[v1.MixerFilter] =
```

Comment:
This should be this field

https://github.com/envoyproxy/data-plane-api/blob/master/envoy/api/v2/route/route.proto#L155

Reply:
I believe this is still WIP @rshriram ?

Reply:
We don't plan to use "Metadata" field in mixer client. We plan to use the new "per_filter_config".

Reply:
@qiwzhang if you are using the per_filter_config, you need to wait for the envoy filter that allows you to simply do routeEntry()->perFilterConfig() 

Please do not overload the semantics of that field to directly retrieve the value yourself. That will cause some trouble. The PR described above should land shortly.